src/stream_io.rs

```diff
@@ -13,10 +11,9 @@ pub unsafe extern "C" fn write_stream<T: Write>(writer: *mut c_void,
                                                 n: c_int)
                                                 -> c_int {
     let bytes = slice::from_raw_parts(c as *const u8, n as usize);
-    if let Ok(_) = (*(writer as *mut T)).write_all(bytes) {
-        return 0;
-    } else {
-        return 1;
+    match (*(writer as *mut T)).write_all(bytes) {
+        Ok(_) => return 0,
+        Err(e) => return e.raw_os_error().unwrap_or(1) as c_int,
```

Comment:
Okay, so looking into it, we actually don't know statically that it won't fail.  If it's from a `std::fs::File` then the error won't fail, but these functions operate on any type that implements `Write` and `Seek` respectively.

So what I'm thinking is that for now we continue to default to 1, but with a TODO to figure out something better.  And then in the future we can try to figure out a way to fully transparently propagate the Rust `Error` through the C++ API (if that's possible).

For `File` that should still result in the correct behavior once I separate out the success flag and the error code.